src/rust-crypto/RoomEncryptor.ts

```diff
@@ -124,27 +125,29 @@ export class RoomEncryptor {
         // We consider a prepareForEncryption as an encryption promise as it will potentially share keys
         // even if it doesn't send an event.
         // Usually this is called when the user starts typing, so we want to make sure we have keys ready when the
-        // message is finally sent. The actual event encryption request will arrive after wait for the prepareForEncryption
-        // promise to resolve, and then do again an ensureEncryptionSession that should be no op as we already share the room key.
+        // message is finally sent.
+        // If `encryptEvent` is invoked before `prepareForEncryption` has completed, the `encryptEvent` call will wait for
+        // `prepareForEncryption` to complete before executing.
+        // The part where `encryptEvent` shares the room key would be no op as it was already performed by prepareForEncryption.
```

Comment:
```suggestion
        // The part where `encryptEvent` shares the room key will then usually be a no-op as it was already performed by `prepareForEncryption`.
```

(It's not quite a no-op: in particular, if any servers are down during `prepareForEncryption`, we'll try to reach them again during `encryptEvent`. Or the device list could change.)